cucumber-core/src/main/java/io/cucumber/core/plugin/TeamCityPlugin.java

```diff
@@ -420,4 +457,39 @@ private String escape(String source) {
                 .replace("]", "|]");
     }
 
+    private static class ComparisonFailureData {
+        private final String expected;
+
+        private final String actual;
+
+        ComparisonFailureData(String expected, String actual) {
+            if (isWrappedWith(expected, '<', '>') && isWrappedWith(actual, '<', '>')) {
```

Comment:
Having studied the different exception messages a bit more I would expect this to be included in the regex. Or perhaps as a different variation of that regex.The right capture group should avoid the need for unwrapping.